assets/tests/components/groupedAutocomplete.test.tsx

```diff
@@ -893,4 +894,105 @@ describe("<GroupedAutocompleteFromSearchTextResults/>", () => {
     expect(vehicleOption.get(vehicleOptions)).toBeInTheDocument()
     expect(runOption.query()).not.toBeInTheDocument()
   })
+
+  test("when an autocomplete option is clicked, should fire event 'onSelectVehicleOption'", async () => {
+    const onSelectVehicleOption = jest.fn()
+    const searchText = "123"
+    const vehicle = vehicleFactory.build()
+
+    jest.mocked(useAutocompleteResults).mockImplementation(((
+      _socket,
+      searchText
+    ) => {
+      if (searchText === searchText) {
+        return {
+          vehicle: [vehicle],
+          operator: [],
+          run: [],
+        }
+      }
+      return {
+        operator: [],
+        run: [],
+        vehicle: [],
+      }
+    }) as typeof useAutocompleteResults)
```

Comment:
suggestion: since we're using `jest.mocked`, I think the `as` is unneeded?

```suggestion
    }))
```